### PR TITLE
New Feature: Python Tools for Connectivity Testing

### PR DESCRIPTION
**Pull request by bdorney**
## Description

Provides functionality for monitoring link health, needed in connectivity testing.

Requires:

https://github.com/cms-gem-daq-project/ctp7_modules/pull/82
https://github.com/cms-gem-daq-project/xhal/pull/102

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Need to monitor status of GBT & VFAT links for connectivity testing.

## How Has This Been Tested?

See above PR's for testing info.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: gempython/tools/amc_user_functions_uhal.py [hunk 1]
```diff
@@ -209,18 +209,28 @@ def printSystemTTCInfo(amc,debug=False):
...
+    scaInfo = nesteddict()
+
     print "--=======================================--"
     print "-> GEM SYSTEM SCA INFORMATION"
     print "--=======================================--"
     print
     nOHs = readRegister(amc,"GEM_AMC.GEM_SYSTEM.CONFIG.NUM_OF_OH")
     scaRdy  = readRegister(amc,"GEM_AMC.SLOW_CONTROL.SCA.STATUS.READY")
     scaCErr = readRegister(amc,"GEM_AMC.SLOW_CONTROL.SCA.STATUS.CRITICAL_ERROR")
+    scaInfo["READY"]=scaRdy
```
Comment (jsturdy) [inline gempython/tools/amc_user_functions_uhal.py]: Merge 225 to 223?

File: gempython/tools/amc_user_functions_uhal.py [hunk 2]
```diff
@@ -209,18 +209,28 @@ def printSystemTTCInfo(amc,debug=False):
...
+    scaInfo["CRITICAL_ERROR"]=scaCErr
```
Comment (jsturdy) [inline gempython/tools/amc_user_functions_uhal.py]: Merge 226 with 224?

File: gempython/tools/amc_user_functions_uhal.py [hunk 3]
```diff
@@ -209,18 +209,28 @@ def printSystemTTCInfo(amc,debug=False):
...
     print("READY             %s0x%08x%s"%(colors.BLUE,scaRdy,colors.ENDC))
     print("CRITICAL_ERROR    %s0x%08x%s"%(colors.RED if scaCErr else colors.GREEN,scaCErr,colors.ENDC))
     for li in range(nOHs):
+        notRdyCnt = readRegister(amc,"GEM_AMC.SLOW_CONTROL.SCA.STATUS.NOT_READY_CNT_OH%d"%(li))
+        scaInfo["NOT_READY_CNT"][li]=notRdyCnt
         print("NOT_READY_CNT_OH%02d %s0x%08x%s"%(li,
                                                  colors.GREEN,
-                                                 readRegister(amc,"GEM_AMC.SLOW_CONTROL.SCA.STATUS.NOT_READY_CNT_OH%d"%(li)),
+                                                 notRdyCnt,
```
Comment (jsturdy) [inline gempython/tools/amc_user_functions_uhal.py]: Merge 231 with 230 and print that in 234?

**Comment by bdorney** [comment]
@jsturdy comments addressed.

**Pull request merged**